packages/fuels-code-gen/src/program_bindings/abigen/bindings/function_generator.rs

```diff
@@ -66,7 +66,17 @@ impl FunctionGenerator {
     }
 
     pub fn tokenized_args(&self) -> TokenStream {
-        let arg_names = self.args.iter().map(|component| &component.field_name);
+        let arg_names = self.args.iter().map(|component| {
+            let field_name = &component.field_name;
+
+            if component.wrap_into {
+                let field_type = &component.field_type;
+
+                quote! {::core::convert::Into::<#field_type>::into(#field_name)}
+            } else {
+                quote! {#field_name}
+            }
```

Comment:
I think this should be replaced with just an associated function on `Component`: 
`component.wraps_into()`

Reply:
resolving as the code changed. See: https://github.com/FuelLabs/fuels-rs/pull/967#issuecomment-1556131371